### ISSUE TITLE
Support changing to custom icons (for groups and entries)

### ISSUE THREAD

**Issue by hpoul** [authpass/authpass#157]
(followup to #68): I have added support for *displaying* existing custom icons, but didn't bother to implement actually adding custom icons. (throws `StateError('not yet supported.');`).

It should be possible to just select a completely custom icon by choosing a Image through the file picker. Maybe also allow selecting existing custom icons which are already embedded in the file.

also: when switching from a custom icon to a predefined icon make sure unused custom icos are removed from the kdbx file (this might be extracted into a separate issue)

**Comment by PatrikKoscelansky** [comment]
Hi, I'd like to give this a try :)

**Comment by hpoul** [comment]
@PatrikKoscelansky nice 👍️ sure, let me know if you need anything. for how to select a file you can take a look at how file attachments work (`entry_details.dart`.. but make sure for custom icons, that only PNGs are loaded. (i don't think any other format is supported)

<img width="536" alt="Screen Shot 2020-09-29 at 13 48 24" src="https://user-images.githubusercontent.com/313066/94554546-8533aa80-025a-11eb-8cae-6bf46481cb9a.png">

**Comment by PatrikKoscelansky** [comment]
@hpoul I started working on this and I have a question: after I load a .png file, I want to create a KdbxCustomIcon, that I'd like to save to the file later. When creating KdbxCustomIcon, should I use KdbxUuid.random() to get new uuid, or should I obtain this uuid from somwhere else?

**Comment by hpoul** [comment]
@PatrikKoscelansky 👍️ yep, `KdbxUuid.random()` is the right way.

**Comment by PatrikKoscelansky** [comment]
hi @hpoul . I'm almost done. I managed to implement adding of new custom icon and changing to them (for entries). Looks like this (looks weird, because I picked weir png file): 

![image](https://user-images.githubusercontent.com/30245783/95462258-8f9b2600-0977-11eb-88d7-56b009ed45c0.png)

During the implementation I came across a problem that was causing error when trying to save newly picked custom icon. I temporarily commented the thing that was causing it in kdbx_object.dart file and I think it should be like this, but maybe I'm wrong:

![image](https://user-images.githubusercontent.com/30245783/95463244-d6d5e680-0978-11eb-896f-9fde1b11ed8f.png)

But most importantly, I can't wrap my head around this: When saving custom group icon (the same way as with entries)...

![image](https://user-images.githubusercontent.com/30245783/95463899-a2165f00-0979-11eb-81d9-aa523f092493.png)

.. I get 'Invalid argument(s)' exception.

**Comment by hpoul** [comment]
@PatrikKoscelansky 👍️ great to see you made progress, regarding `set customIcon`: i think it should be an `} else {` .. otherwise, with your current code it would not be possible to remove a `customIcon`. regarding the invalid argument: not sure either, can you show me the full stacktrace? Although I'm not sure it's valid to set it to `null` .. i think I should just keep both. at least KeePassXC seems to do the same:

```
        <IconID>0</IconID>
        <CustomIconUUID>q1TE/641TayCc+XiNFLKoQ==</CustomIconUUID>
```

(IconID 0 is the default "key" icon)

**Comment by hpoul** [comment]
implemented in #187 🥳
